lib/trino-parquet/src/test/java/io/trino/parquet/writer/NullsProvider.java

```diff
@@ -79,6 +79,26 @@ Optional<boolean[]> getNulls(int positionCount)
 
     abstract Optional<boolean[]> getNulls(int positionCount);
 
+    Optional<boolean[]> getNulls(int positionCount, Optional<boolean[]> requiredNulls)
+    {
+        Optional<boolean[]> nulls = getNulls(positionCount);
+        if (requiredNulls.isEmpty()) {
+            return nulls;
+        }
+        if (nulls.isEmpty()) {
+            return requiredNulls;
+        }
+
+        boolean[] nullPositions = nulls.get();
+        boolean[] requiredPositions = requiredNulls.get();
```

Comment:
Maybe name this `forcedNulls` since "required positions" reads as "non-null positions"